eclass/python-single-r1.eclass

```diff
@@ -202,7 +202,25 @@ _python_single_set_globals() {
 		optflags+=,${flags[0]/%/(+)}
 	else
 		# Multiple supported implementations; honor PYTHON_SINGLE_TARGET.
-		IUSE+=" ${flags[*]}"
+		# Check if the ebuild only supports py2 impls, if so
+		# we will later '+' to CPython 2.7
+		local py2_only=true
+		for i in "${_PYTHON_SUPPORTED_IMPLS[@]}"; do
+			if python_is_python3 "${i}"; then
+				py2_only=false
+				break
+			fi
+		done
+
+		if [[ ${py2_only} == true ]]; then
+			# Only py2 impls are in _PYTHON_SUPPORTED_IMPLS, we proceed
+			# to add +python_single_target_python2_7 to IUSE
+			IUSE+=" ${flags[*]/python_single_target_python2_7/+python_single_target_python2_7}"
```

Comment:
I don't like the long form here but I guess I can live with it if there's no other way ;-P.